doc/source/user/basics.indexing.rst

```diff
@@ -194,8 +194,16 @@ concepts to remember include:
 - If the selection tuple has all entries ``:`` except the
   *p*-th entry which is a slice object ``i:j:k``,
   then the returned array has dimension *N* formed by
-  concatenating the sub-arrays returned by integer indexing of
-  elements *i*, *i+k*, ..., *i + (m - 1) k < j*,
+  stacking, along the *p*-th axis, the sub-arrays returned by integer
+  indexing of elements *i*, *i+k*, ..., *i + (m - 1) k < j*. This means
+  that, for example, with the array::
+
+      >>> x = np.arange(24).reshape(2, 4, 3)
+
+  The following are equivalent::
+
+      >>> x[:, 1:3, :]
+      >>> np.stack([x[:, i, :] for i in range(1, 3)], axis=1)
```

Comment:
The main issue here is that the resulting arrays have the same values, but the operations are not equivalent as the first returns a view whereas the `stack` returns a new array.

I understand the desire to demonstrate the indexing with an example, but I don't think this is a good example to add in the context of basic indexing.

Reply:
Ah, you're right, the return values are not truly equivalent.

I can add a further explanation of that caveat (and fix the failing doctests), or are you saying it's too complicated for this part of the doc? Should I remove the example altogether?

Reply:
I don't have a strong preference, but my vote would be to drop the example to avoid confusion. The wording update (concatenating subarrays -> stacking along axis) seems fine, though YMMV whether it improves clarity